lib/fika/compiler/type_checker.ex

```diff
@@ -31,25 +33,44 @@ defmodule Fika.Compiler.TypeChecker do
   def infer({:function, _line, {name, args, _type, exprs}}, env) do
     Logger.debug("Inferring type of function: #{name}")
 
-    env =
-      env
-      |> Map.put(:scope, %{})
-      |> Map.put(:has_effect, false)
-      |> add_args_to_scope(args)
+    mfa = {env[:module_name], name, length(args)}
 
-    case infer_block(env, exprs) do
-      {:ok, type, env} ->
-        type =
-          if env[:has_effect] do
-            %T.Effect{type: type}
-          else
-            type
-          end
+    result =
+      if latest_call = env[:latest_called_function] do
+        FunctionDependencies.set_function_dependency(
```

Comment:
For each subsequent call in the stack, we have both the mfa for the last called function and for the current one. This implies that `:latest_called_function` depends on `mfa`, so we can set this on the graph